packages/roosterjs-content-model-markdown/lib/modelToMarkdown/creators/createMarkdownParagraph.ts

```diff
@@ -0,0 +1,47 @@
+import { MarkdownHeadings } from '../../constants/headings';
+import type { ContentModelParagraph, ContentModelText } from 'roosterjs-content-model-types';
+
+/**
+ * @internal
+ */
+export function createMarkdownParagraph(paragraph: ContentModelParagraph): string {
+    const { segments } = paragraph;
+    let markdownString = '';
+    for (const segment of segments) {
+        switch (segment.segmentType) {
+            case 'Text':
+                markdownString += textProcessor(segment);
+                break;
+            case 'Image':
+                markdownString += `![${segment.alt}](${segment.src})`;
+                break;
+            case 'Br':
+                markdownString += '\n';
```

Comment:
For paragraph under list, we need to indent
For paragraph under block quote, we need to add "|"
For paragraph under table, we should not break table structure

Reply:
So I think we may need a context object to remember current state, and when create new line, what kind of related things to add, especially these things combined together.